seed/views/v3/data_quality_checks.py

```diff
@@ -0,0 +1,168 @@
+# !/usr/bin/env python
+# encoding: utf-8
+"""
+:copyright (c) 2014 - 2020, The Regents of the University of California, through Lawrence Berkeley National Laboratory (subject to receipt of any required approvals from the U.S. Department of Energy) and contributors. All rights reserved.  # NOQA
+:author
+"""
+
+import csv
+
+from celery.utils.log import get_task_logger
+from django.http import JsonResponse, HttpResponse
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import viewsets, status
+from rest_framework.decorators import action
+from unidecode import unidecode
+
+from seed.data_importer.tasks import do_checks
+from seed.decorators import ajax_request_class
+from seed.lib.superperms.orgs.decorators import has_perm_class
+from seed.models.data_quality import DataQualityCheck
+from seed.utils.api import api_endpoint_class
+from seed.utils.api_schema import AutoSchemaHelper
+from seed.utils.cache import get_cache_raw
+
+logger = get_task_logger(__name__)
+
+
+class DataQualityCheckViewSet(viewsets.ViewSet):
+    """
+    Handles Data Quality API operations within Inventory backend.
+    (1) Post, wait, get…
+    (2) Respond with what changed
+    """
+
+    # Remove lookup_field once data_quality_check_id is used and "pk" can be used
+    lookup_field = 'organization_id'
+
+    @swagger_auto_schema(
+        manual_parameters=[
+            AutoSchemaHelper.base_field(
+                "organization_id",
+                "IN_PATH",
+                "Organization ID - identifier used to specify a DataQualityCheck",
+                True,
+                "TYPE_INTEGER"
+            )
+        ],
+        request_body=AutoSchemaHelper.schema_factory(
+            {
+                'property_state_ids': ['integer'],
+                'taxlot_state_ids': ['integer'],
+            },
+            description='An object containing IDs of the records to perform'
+                        ' data quality checks on. Should contain two keys- '
+                        'property_state_ids and taxlot_state_ids, each of '
+                        'which is an array of appropriate IDs.',
+        ),
+        responses={
+            200: AutoSchemaHelper.schema_factory({
+                'num_properties': 'integer',
+                'num_taxlots': 'integer',
+                'progress_key': 'string',
+                'progress': {},
+            })
+        }
+    )
+    @api_endpoint_class
+    @ajax_request_class
+    @has_perm_class('requires_member')
+    @action(detail=True, methods=['POST'])
+    def start(self, request, organization_id):
+        """
+        This API endpoint will create a new data_quality check process in the background,
+        on potentially a subset of properties/taxlots, and return back a query key
+        """
+        body = request.data
+        property_state_ids = body['property_state_ids']
+        taxlot_state_ids = body['taxlot_state_ids']
```

Comment:
@macintoshpie Do you think these should be `_view_ids` instead? I'm leaning that way but wanted your thoughts.

Reply:
I'm honestly not sure. I took a look at how the api is used on the front end ([here](https://github.com/SEED-platform/seed/blob/4829ad94af7b7391e6ce28f8873c6d0b39bcdc94/seed/static/seed/js/controllers/inventory_list_controller.js#L404)) and it seems kind of trivial to make the change there. I'm not totally sure how, on the backend, you find the correct property states given a list of property views, but if that's easy as well then 🤷 .

I guess this goes back to our discussion of whether an api user is aware of and cares about <Inventory>States.

Reply:
Agree - both wouldn't take too much effort to change. I'll tag you for re-review.